### PR TITLE
wishart_cholesky signatures

### PR DESCRIPTION
**Pull request by spinkney**
#### Submission Checklist

- [x] Run unit tests
- Documentation
    - [x] If a user-facing facing change was made, the documentation PR is here: https://github.com/stan-dev/docs/pull/526

Adds `wishart_cholesky_lpdf` and `wishart_cholesky_rng`.

## Release notes

Added `wishart_cholesky_lpdf` and `wishart_cholesky_rng` distribution functions.

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by WardBrian** [comment]
@spinkney it looks like you need to run `make format; dune promote` to get CI to run. The code itself looks good

**Review by WardBrian — APPROVED** [review]
LGTM. Will merge when tests pass

**Comment by WardBrian** [comment]
Failures look like it's due to the fact that we implictly add a `_log` function even though this has been deprecated, and I'm guessing the stan-math PR didn't add one?

**Comment by spinkney** [comment]
that's right, I didn't add one

**Comment by WardBrian** [comment]
Yeah, we honestly shouldn’t be adding new _log signatures at this point since we’re going to remove them all so soon. Tomorrow I’ll look at re-writing how we add these signatures to let us not add it for things that didn’t have it before.

**Comment by spinkney** [comment]
will this add an `lupdf` signature?

**Comment by WardBrian** [comment]
All lpdf signatures are implicitly lupdf signatures as well I believe

**Comment by spinkney** [comment]
ok, honestly I wasn't sure how that was going to get done b/c they aren't in stan-math

**Comment by WardBrian** [comment]
It’s not a bad idea to add some calls of it to the test file. The way we handle lupdfs is honestly pretty poorly, when typechecking we detect unnormalized suffixes and temporarily replace them with the normalized versions to see if the signatures work

**Comment by spinkney** [comment]
@WardBrian it still added `_log` :(

**Comment by WardBrian** [comment]
It looks like it just needs to have the test output updated

File: src/middle/Stan_math_signatures.ml
Comment (WardBrian) [inline src/middle/Stan_math_signatures.ml:341]: All of these changes introduced in the merging are wrong - this should only be a 1-line diff of just `  ; ([Lpdf], "wishart_cholesky", [DMatrix; DReal; DMatrix], SoA)`

Comment (spinkney) [reply src/middle/Stan_math_signatures.ml]: yes

**Comment by WardBrian** [comment]
I think you still removed the `log` suffix from the old Wishart in the merge, which is why the latest tests failed

**Comment by WardBrian** [comment]
This is now passing. Once the docs PR gets approved we can merge - not confident enough in the theory to review the docs myself

**Pull request merged**
